### PR TITLE
Standardize the date format input

### PR DESCRIPTION
**Pull request by faizalzakaria**
@amree , this is what I meant in the #16 

p/s: This is just the spec.

### CODE REVIEW

File: spec/support/date_format_shared_examples.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+shared_examples :date_format do
```
Comment (faizalzakaria) [inline spec/support/date_format_shared_examples.rb:1]: this test, @amree . e.g: Air Asia & Firefly failed 2 of these test.

**Comment by faizalzakaria** [comment]
I extended it to support more than just ISO8601. As seems its easy to just hold Date object and manipulate the date as needed by the target.

**Comment by amree** [comment]
:+1: with support for more than one date format, let me try running it locally first. It's hard too read online when there're too many changes in one pull request (which doesn't matter because it's necessary).

**Comment by faizalzakaria** [comment]
@amree , You can ignore the `spec/cassettes/**/*`.

File: Rakefile [hunk 2]
```diff
@@ -3,5 +3,5 @@ require "rspec/core/rake_task"
 
 RSpec::Core::RakeTask.new
 
-task :default => :spec
-task :test => :spec
+task default: :spec
+task test: :spec
```
Comment (amree) [inline Rakefile:7]: Thanks for these two lines

File: lib/teecket/air_asia.rb [hunk 3]
```diff
@@ -21,15 +26,15 @@ def get
     req.add_field("Channel", "mobile-touch-app")
 
     req.body = URI.encode_www_form([
-      ["type", "classic"],
+      %w(type classic),
```
Comment (amree) [inline lib/teecket/air_asia.rb]: How about keeping it as array element as others are using array, for consistency

Comment (faizalzakaria) [inline lib/teecket/air_asia.rb]: agree.

File: spec/support/date_format_shared_examples.rb (hunk 1 above)
Comment (amree) [inline spec/support/date_format_shared_examples.rb:1]: Just want to give you heads up. AirAsia updated their mobile app which now requires BIG ID credential before logging in. I'm also getting HTTPForbidden error using the master branch. I'm still setting up my python `mitmproxy` but those two incidents are highly possibly related.

**Comment by amree** [comment]
After thinking for a while, I think I'm going to merge this pull request and treat whatever's not working as bug so that this gem can move forward (especially in updating AirAsia outdated scrapping method and adding Rayani Air to the list). What do you think?

**Comment by faizalzakaria** [comment]
sounds good @amree

**Comment by amree** [comment]
Merged. Thanks a lot for the pull request. Really appreciate it.

**Pull request merged**